include/deal.II/lac/constraint_matrix.templates.h

```diff
@@ -2269,8 +2274,13 @@ ConstraintMatrix::distribute_local_to_global (
   // an array in any case since we cannot know about the actual data type in
   // the ConstraintMatrix class (unless we do cast). This involves a little
   // bit of logic to determine the type of the matrix value.
-  std::vector<size_type> &cols = scratch_data->columns;
-  std::vector<number>     &vals = scratch_data->values;
+  std::vector<size_type> &cols          = scratch_data->columns;
+  std::vector<number>    &vals          = scratch_data->values;
+  // create arrays for writing into the vector as well
+  std::vector<size_type> &rows          = scratch_data->vector_indices;
```

Comment:
`rows` suggests that these are matrix rows (because we already have `cols`). Maybe rename `vector_elements` or similar?